wrappers/react/test/hotTable.spec.tsx

```diff
@@ -355,4 +355,81 @@ describe('Editor configuration using React components', () => {
       activeEditor.close();
     }
   });
+
+  it('should use the correct renderer inside HotTable component depends on its mount state', async () => {
+    let hotTableInstanceRef = React.createRef();
+
+    class WrapperComponent extends React.Component<any, any> {
+      state = {
+        renderer: false,
+      }
+
+      render() {
+        return (
+          <HotTable licenseKey="non-commercial-and-evaluation"
+                    id="test-hot"
+                    data={createSpreadsheetData(3, 3)}
+                    width={300}
+                    height={300}
+                    rowHeights={23}
+                    colWidths={50}
+                    init={function () {
+                      mockElementDimensions(this.rootElement, 300, 300);
+                    }}
+                    ref={hotTableInstanceRef}>
+            {this.state.renderer ?  <RendererComponent hot-renderer /> : null}
+          </HotTable>
+        );
+      };
+    }
+
+    const wrapperComponentInstance = mountComponentWithRef((
+      <WrapperComponent/>
+    ));
+
+    let hotInstance = (hotTableInstanceRef.current as any).hotInstance;
+
+    await act(async() => {
+      hotInstance.selectCell(0, 0);
+    });
+
+    {
+      const activeRenderer = hotInstance.getCellRenderer(0, 0);
+
+      expect(activeRenderer.name).toBe('textRenderer');
+    }
+
+    await act(async() => {
+      wrapperComponentInstance.setState({ renderer: true });
+    });
+
+    await sleep(100);
+
+    await act(async() => {
+      hotInstance.selectCell(0, 0);
+    });
+
+    {
+      const activeRenderer = hotInstance.getCellRenderer(0, 0);
+
+      expect(activeRenderer.name).toBe('');
```

Comment:
```suggestion
      expect(activeRenderer.name).toBe('__internalRenderer');
```
To make sure that the test doesn't pass when an anonymous function is passed to the `renderer`, I would name it `__internalRenderer`, for example. 
You would have to change this line to make it work
```diff
diff --git a/wrappers/react/src/hotTableClass.tsx b/wrappers/react/src/hotTableClass.tsx
index 82288baab..ef2ef0365 100644
--- a/wrappers/react/src/hotTableClass.tsx
+++ b/wrappers/react/src/hotTableClass.tsx
@@ -225,7 +225,7 @@ class HotTableClass extends React.Component<HotTableProps, {}> {
   getRendererWrapper(rendererElement: React.ReactElement): typeof Handsontable.renderers.BaseRenderer | any {
     const hotTableComponent = this;
 
-    return function (instance, TD, row, col, prop, value, cellProperties) {
+    return function __internalRenderer(instance, TD, row, col, prop, value, cellProperties) {
       const renderedCellCache = hotTableComponent.getRenderedCellCache();
 
       if (renderedCellCache.has(`${row}-${col}`)) {

```

Reply:
done